aten/src/ATen/native/quantized/cpu/qembeddingbag_prepack.cpp

```diff
@@ -299,16 +312,34 @@ Tensor _qembeddingbag_nbit_prepack_helper(
 
 #ifdef USE_FBGEMM
   if (!optimized_qparams) {
-    at::parallel_for(
-      0, embedding_rows, 1, [&](int32_t start_idx, int32_t end_idx) {
-        for (int64_t row = start_idx; row < end_idx; ++row) {
-          fbgemm::FloatOrHalfToFusedNBitRowwiseQuantizedSBHalf<float>(
-            bit_width, weight_data + row * embedding_cols, 1,
-            embedding_cols, output_data + row * output_shape[1]);
-        }
-      });
+    if (weight.scalar_type() == at::ScalarType::Half) {
+      const auto weight_data = static_cast<fbgemm::float16*>(weight.data_ptr());
+      at::parallel_for(
+        0, embedding_rows, 1, [&](int32_t start_idx, int32_t end_idx) {
+          for (int64_t row = start_idx; row < end_idx; ++row) {
+            fbgemm::FloatOrHalfToFusedNBitRowwiseQuantizedSBHalf<fbgemm::float16>(
+              bit_width, weight_data + row * embedding_cols, 1,
+              embedding_cols, output_data + row * output_shape[1]);
+          }
+        });
+    }
+    else {
+      const auto weight_data = weight.data_ptr<float>();
+      at::parallel_for(
+        0, embedding_rows, 1, [&](int32_t start_idx, int32_t end_idx) {
+          for (int64_t row = start_idx; row < end_idx; ++row) {
+            fbgemm::FloatOrHalfToFusedNBitRowwiseQuantizedSBHalf<float>(
+              bit_width, weight_data + row * embedding_cols, 1,
+              embedding_cols, output_data + row * output_shape[1]);
+          }
+        });
+    }
+
+
   } else {
 #endif // USE_FBGEMM
+    TORCH_CHECK(weight.scalar_type() == at::ScalarType::Float, "'embedding_bag_byte_prepack' only support float32, when not use fbgemm.");
```

Comment:
Can we add a fallback path for fp16 as well that works without fbgemm? It can just cast the input and do the regular computaion

Reply:
pushed a new version with fallback logic (cast fp16 to fp32) when fbgemm is not enabled.